packages/block-editor/src/components/block-edit/test/edit.js

```diff
@@ -84,6 +86,47 @@ describe( 'Edit', () => {
 		expect( editElement ).toHaveClass( 'my-class' );
 	} );
 
+	it( 'should combine the default class name with a variation one', () => {
+		const edit = ( { className } ) => (
+			<div data-testid="foo-bar" className={ className } />
+		);
+
+		registerBlockType( 'core/test-block', {
+			edit,
+			save: noop,
+			category: 'text',
+			title: 'block title',
+			attributes: {
+				fruit: {
+					type: 'string',
+					default: 'Apples',
+				},
+			},
+		} );
+
+		registerBlockVariation( 'core/test-block', {
+			name: 'variation',
+			title: 'block variation title',
+			attributes: {
+				fruit: {
+					type: 'string',
+					default: 'Bananas',
+				},
```

Comment:
I just noticed that this looks like a _block_ attribute definition (with a type and default). However, a variation is defined by the concrete values for its block attributes, i.e.

```suggestion
				fruit: 'Bananas',
```

It should be possible to verify that that's correct by using the shorthand notation for `isActive`, i.e. `[ 'Bananas']`.